src/test/scala/org/broadinstitute/dsde/workbench/sam/service/MockUserServiceBuilder.scala

```diff
@@ -107,7 +107,17 @@ case class MockUserServiceBuilder() extends IdiomaticMockito {
     mockUserService.getUser(eqTo(samUser.id), any[SamRequestContext]) answers ((_: WorkbenchUserId) => IO(Option(samUser)))
     mockUserService.deleteUser(eqTo(samUser.id), any[SamRequestContext]) returns IO(())
     mockUserService.updateUserCrud(eqTo(samUser.id), any[AdminUpdateUserRequest], any[SamRequestContext]) answers (
-      (_: WorkbenchUserId, r: AdminUpdateUserRequest) => IO(Option(samUser.copy(email = r.email.get)))
+      (_: WorkbenchUserId, r: AdminUpdateUserRequest) => {
+        val newAzureB2CId = if (r.azureB2CId.contains(AzureB2CId(""))) None else if (r.azureB2CId.isDefined) r.azureB2CId else samUser.azureB2CId
+        val newGoogleSubjectId = if (r.googleSubjectId.contains(GoogleSubjectId(""))) None else if (r.googleSubjectId.isDefined) r.googleSubjectId else samUser.googleSubjectId
+        val newUser = samUser.copy(
+          email = r.email.getOrElse(samUser.email),
+          enabled = r.enabled.getOrElse(samUser.enabled),
+          azureB2CId = newAzureB2CId,
+          googleSubjectId = newGoogleSubjectId
+        )
+        IO(Option(newUser))
+      }
```

Comment:
Following up on my comment above I am not a huge fan of having to fake the underlying implementation in these mocks, because we end up testing the mock implementation in the routes test rather than the underlying implementation. Although other test suites should be covering the underlying impl while the routes cover more general request -> response patterns.